shared/src/sources/balancer/pool_fetching.rs

```diff
@@ -105,19 +108,48 @@ impl BalancerPoolEvaluating for WeightedPool {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct AmplificationParameter {
+    factor: U256,
+    precision: U256,
+}
+
+impl AmplificationParameter {
+    pub fn new(factor: U256, precision: U256) -> Self {
+        Self { factor, precision }
+    }
+
+    /// This is the format used to pass into smart contracts.
+    pub fn as_u256(&self) -> U256 {
+        self.factor
+    }
+
+    /// This is the format used to pass along to HTTP solver.
+    pub fn as_big_rational(&self) -> Result<BigRational> {
+        // In order to circumvent the possibility of panic from BigRational::new,
+        // we check for zero denominator and return result
+        ensure!(!self.precision.is_zero(), "Zero precision not allowed");
```

Comment:
```suggestion
        // We can assert that the precision is non-zero as we check when constructing
        // new `AmplificationParameter` instances that this invariant holds, and we don't
        // allow modifications of `self.precision` such that it could become 0.
        debug_assert!(!self.precision.is_zero());
```